src/components/node-list/node-list-tree.js

```diff
@@ -184,19 +211,25 @@ const TreeListProvider = ({
       focusModeIcon = isFocusedModularPipeline ? FocusModeIcon : null;
     }
 
+    const isModularPipelineCollapsed = !expanded.includes(node.id);
+
+    const data = getModularPipelineRowData({
+      ...node,
+      focusModeIcon,
+      disabled: focusMode && !isOnFocusedModePath(focusMode.id, node.id),
+      focused: isFocusedModularPipeline,
+      highlight: slicedPipeline.includes(node.id) && isModularPipelineCollapsed,
+    });
+
     return (
       <NodeListTreeItem
-        data={getModularPipelineRowData({
-          ...node,
-          focusModeIcon,
-          disabled: focusMode && !isOnFocusedModePath(focusMode.id, node.id),
-          focused: isFocusedModularPipeline,
-        })}
+        data={data}
         onItemMouseEnter={onItemMouseEnter}
         onItemMouseLeave={onItemMouseLeave}
         onItemChange={onItemChange}
         onItemClick={onItemClick}
-        key={node.id}
+        key={uniqueId(node.id)}
```

Comment:
we need this here because there's warning in the browser
<img width="1710" alt="Screenshot 2024-08-20 at 15 15 58" src="https://github.com/user-attachments/assets/97c35277-b322-4c59-92fc-c18190181303">